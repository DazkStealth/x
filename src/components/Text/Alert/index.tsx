import React from 'react';
import styled, { useTheme } from 'styled-components';

const Alert = ({
  level = 'info',
  children,
}: React.PropsWithChildren<{
  level?: 'info' | 'success' | 'warning' | 'error'
}>) => {
  const theme = useTheme();
  return (
    <AlertText
      $background={theme.color.background[level]}
      $color={theme.color.text[level]}
    >
      {children}
    </AlertText>
  );
}

const AlertText = styled.p<{ $color: string; $background: string; }>`
  padding: 15px;
  background: ${({ $background }) => $background};
  color: ${({ $color }) => $color};
  word-break: break-all;
  margin-bottom: 15px;
  font-size: 14px;
  border-radius: 5px;
`;


export default Alert;
