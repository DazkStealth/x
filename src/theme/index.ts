import { keyframes } from 'styled-components';

export interface Theme {
  font: {
    primary: string;
  }
  color: {
    background: {
      body: string;
      bottomMenu: string;
      bottomMenuItem: string;
      loadingLogo: string;
      buttonPrimary: string;
      input: string;
      info: string;
      error: string;
      success: string;
      warning: string;
    },
    text: {
      body: string;
      bottomMenuItem: string;
      bottomMenuItemActive: string;
      loadingLogo: string;
      buttonPrimary: string;
      input: string;
      info: string;
      error: string;
      success: string;
      warning: string;
    },
    border: {
      input: string;
      bottomMenuItemBottomActive: string;
    }
  }
}

export const defaultTheme: Theme = {
  font: {
    primary: '"Roboto", sans',
  },
  color: {
    background: {
      body: '#24142f',
      bottomMenu: 'rgba(50,50,66,0.7)',
      bottomMenuItem: '#444d55',
      loadingLogo: '#fff',
      buttonPrimary: '#c63bea',
      input: '#fff',
      info: '#BEF',
      error: '#D8000C',
      success: '#DFF2BF',
      warning: '#FEEFB3',
    },
    text: {
      body: '#fff',
      bottomMenuItem: '#fff',
      bottomMenuItemActive: '#444d55',
      loadingLogo: '#997cfa',
      buttonPrimary: '#fff',
      input: '#000',
      info: '#059',
      error: '#FFBABA',
      success: '#270',
      warning: '#9F6000',
    },
    border: {
      input: '#000',
      bottomMenuItemBottomActive: '#fff',
    }
  },
};

const skeleton = keyframes`
  0% {
    background-color: hsl(200, 20%, 80%);
  }
  100% {
      background-color: hsl(200, 20%, 95%);
  }
`;


const pulse = keyframes`
  0% {
    transform: scale(1);
  }
  50% {
    transform: scale(2);
  }
  100% {
    transform: scale(1);
  }
`;

export const animation = {
  skeleton,
  pulse,
}
