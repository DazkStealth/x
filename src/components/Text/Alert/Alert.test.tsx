import renderer, { ReactTestRendererJSON } from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';

// components
import Alert from './';

// theme
import { defaultTheme } from '../../../theme';

describe('<Alert />', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={defaultTheme}>
          <Alert>default</Alert>
          <Alert level="info">info</Alert>
          <Alert level="warning">warning</Alert>
          <Alert level="success">success</Alert>
          <Alert level="error">error</Alert>
          {/* @ts-expect-error: pass wrong level for test purpose */}
          <Alert level="non-exsisting">non-existing</Alert>
        </ThemeProvider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();

    const treeElements = tree as ReactTestRendererJSON[];
    expect(treeElements[0].children?.length).toBe(1);
    expect(treeElements[0].children?.[0]).toBe('default');
    expect(treeElements[0].type).toBe('p');
    expect(treeElements[0]).toHaveStyleRule('color', defaultTheme.color.text.info);
    expect(treeElements[0]).toHaveStyleRule('background', defaultTheme.color.background.info);
    expect(treeElements[1]).toHaveStyleRule('color', defaultTheme.color.text.info);
    expect(treeElements[1]).toHaveStyleRule('background', defaultTheme.color.background.info);
    expect(treeElements[2]).toHaveStyleRule('color', defaultTheme.color.text.warning);
    expect(treeElements[2]).toHaveStyleRule('background', defaultTheme.color.background.warning);
    expect(treeElements[3]).toHaveStyleRule('color', defaultTheme.color.text.success);
    expect(treeElements[3]).toHaveStyleRule('background', defaultTheme.color.background.success);
    expect(treeElements[4]).toHaveStyleRule('color', defaultTheme.color.text.error);
    expect(treeElements[4]).toHaveStyleRule('background', defaultTheme.color.background.error);
    expect(treeElements[5]).toHaveStyleRule('color', undefined);
    expect(treeElements[5]).toHaveStyleRule('background', undefined);
  });
});

