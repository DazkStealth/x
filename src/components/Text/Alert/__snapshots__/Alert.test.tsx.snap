// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Alert /> renders correctly 1`] = `
[
  .c0 {
  padding: 15px;
  background: #BEF;
  color: #059;
  word-break: break-all;
  margin-bottom: 15px;
  font-size: 14px;
  border-radius: 5px;
}

<p
    background="#BEF"
    className="c0"
    color="#059"
  >
    default
  </p>,
  .c0 {
  padding: 15px;
  background: #BEF;
  color: #059;
  word-break: break-all;
  margin-bottom: 15px;
  font-size: 14px;
  border-radius: 5px;
}

<p
    background="#BEF"
    className="c0"
    color="#059"
  >
    info
  </p>,
  .c0 {
  padding: 15px;
  background: #FEEFB3;
  color: #9F6000;
  word-break: break-all;
  margin-bottom: 15px;
  font-size: 14px;
  border-radius: 5px;
}

<p
    background="#FEEFB3"
    className="c0"
    color="#9F6000"
  >
    warning
  </p>,
  .c0 {
  padding: 15px;
  background: #DFF2BF;
  color: #270;
  word-break: break-all;
  margin-bottom: 15px;
  font-size: 14px;
  border-radius: 5px;
}

<p
    background="#DFF2BF"
    className="c0"
    color="#270"
  >
    success
  </p>,
  .c0 {
  padding: 15px;
  background: #D8000C;
  color: #FFBABA;
  word-break: break-all;
  margin-bottom: 15px;
  font-size: 14px;
  border-radius: 5px;
}

<p
    background="#D8000C"
    className="c0"
    color="#FFBABA"
  >
    error
  </p>,
  .c0 {
  padding: 15px;
  word-break: break-all;
  margin-bottom: 15px;
  font-size: 14px;
  border-radius: 5px;
}

<p
    className="c0"
  >
    non-existing
  </p>,
]
`;
